### ISSUE TITLE
jacktrip 1.2.1 for Windows does not connect to server

### ISSUE THREAD

**Issue by jamestparker** [jacktrip/jacktrip#139]
jackd on server - 1.9.14
jackd on local machine - 1.9.11

jacktrip on server - 1.2.1
jacktrip on local machine - 1.2.1

Hi everyone,
I'm sorry that I don't know the issue template for this project, but I downloaded the new version of Jacktrip for Windows (1.2.1) and tried to connect to my server.  Doing this fails repeatedly (which jacktrip 1.2 does not do).  I printed out the verbose error message, and it looks like the new jacktrip is communicating with Jack incorrectly.  Every time I try to connect, the peer buffer size is "random", the peer audio bit resolution is "random", and the peer sample rate is simply incorrect.  This occurs in "step 6" of the connection process.  What's interesting is that I still receive a connection from the peer, but it terminates immediately after step 8.

Below are two sample outputs from step 6, followed by a complete error message.

**Example 1**

    step 6 before Setup Audio Packet buffer, Full Packet buffer, Redundancy Variables

    jacktrip_main before app.exec()
        UdpDataProtocol:run0 before setRealtimeProcessPriority()
    100ms
        UdpDataProtocol:run1 before mJackTrip->checkPeerSettings()
    ERROR: Peer Buffer Size is  : 38205
           Local Buffer Size is : 128
    Make sure both machines use same buffer size
    
    ERROR: Peer Sampling Rate is   : 22050
           Local Sampling Rate is  : 48000
    Make sure both machines use the same Sampling Rate
    
    ERROR: Peer Audio Bit Resolution is  : 23
           Local Audio Bit Resolution is : 16
    Make sure both machines use the same Bit Resolution
    
    
**Example 2**

    step 6
    jacktrip_main before app.exec()
    100ms  100ms
        UdpDataProtocol:run1 before mJackTrip->checkPeerSettings()
    ERROR: Peer Buffer Size is  : 3926
           Local Buffer Size is : 128
    Make sure both machines use same buffer size

    ERROR: Peer Sampling Rate is   : 22050
           Local Sampling Rate is  : 48000
    Make sure both machines use the same Sampling Rate
    
    ERROR: Peer Audio Bit Resolution is  : 9
           Local Audio Bit Resolution is : 16
    Make sure both machines use the same Bit Resolution
    
**Complete message**

    Verbose mode
    Settings:startJackTrip before new JackTrip
    Settings:startJackTrip before mJackTrip->setBindPorts
    Settings:startJackTrip before mJackTrip->setPeerPorts
    Settings:startJackTrip before mJackTrip->startProcess
    step 1
      JackTrip:startProcess before checkIfPortIsBinded(mReceiverBindPort)
      JackTrip:startProcess before checkIfPortIsBinded(mSenderBindPort)
      JackTrip:startProcess before setupAudio
      JackTrip:setupAudio before new JackAudioInterface
      JackTrip:setupAudio before mAudioInterface->setup
    Setting JACK Process Callback...
    SUCCESS
    
    The Sampling Rate is: 48000
    
    The Audio Buffer Size is: 128 samples
                          or: 512 bytes
    
    The Number of Channels is: 2
    
    The RTAudio device ID is: 0
    
    Using UDP Protocol
    
    step 2C client only
      JackTrip:startProcess case CLIENTTOPINGSERVER before clientPingToServerStart
    Connecting to TCP Server...
    TCP Socket Connected to Server!
    Port sent to Server
    Reading UDP port from Server...
    Ready To Read From Socket!
    Connection Succesfull!
    Server Address set to: xxx.xx.xx.xx Port: xxxxx

        UdpDataProtocol:run1 before bindSocket(UdpSocket)
      JackTrip:startProcess before mDataProtocolReceiver->start
      JackTrip:startProcess before mDataProtocolSender->start
    step 3
    step 4
    UDP Socket Receiving in Port: 4464
    
        UdpDataProtocol:run1 before Setup Audio Packet buffer, Full Packet buffer, Redundancy Variables
    step 5
      JackTrip:startProcess before mAudioInterface->startProcess
        UdpDataProtocol:run1 before setRealtimeProcessPriority()
        UdpDataProtocol:run0 before Setup Audio Packet buffer, Full Packet buffer, Redundancy Variables
        UdpDataProtocol:run0 before setRealtimeProcessPriority()
    step 6
    jacktrip_main before app.exec()
        UdpDataProtocol:run1 before !UdpSocket.hasPendingDatagrams()
    Waiting for Peer...
    100ms  100ms
        UdpDataProtocol:run1 before mJackTrip->checkPeerSettings()
    ERROR: Peer Buffer Size is  : 40375
           Local Buffer Size is : 128
    Make sure both machines use same buffer size
    
    ERROR: Peer Sampling Rate is   : 22050
           Local Sampling Rate is  : 48000
    Make sure both machines use the same Sampling Rate
    
    ERROR: Peer Audio Bit Resolution is  : 25
           Local Audio Bit Resolution is : 16
    Make sure both machines use the same Bit Resolution
    
    step 7
        UdpDataProtocol:run1 before mJackTrip->parseAudioPacket()
    Received Connection from Peer!
    Stopping JackTrip...
    step 8
    JackTrip Processes STOPPED!
    
    Exiting JackTrip...

Thank you!

**Comment by psiborg112** [comment]
What platform are you using for the server? (Also windows?) Can't seem to recreate it here, but only have one windows machine available so I was on linux for the server.

The random values suggest that something is going wrong with the header. What command line options are you running with?

**Comment by SynthiaPayne** [comment]
Recently there was a Windows user reporting similar random values in 
sample rate, etc. Apparently it is related to printer sharing being set 
to on. I'm not sure if this was triggered by something in the header as 
Aaron mentioned below.

Here's what they did to correct it:

1. In Windows Search Type: sharing
2. Choose “Manage Advanced Sharing”
3. Turn off printer and file sharing for both public and private

If needed, when jacktrip session is finished – turn printer and file 
sharing back on.

synth

**Comment by jamestparker** [comment]
@psiborg112     The server uses a CentOS Linux distribution, and I ran jacktrip on my Windows 10 with -C  and -q 12

@SynthiaPayne  That solution worked perfectly, thank you so much!

**Comment by psiborg112** [comment]
Thanks for the info Synthia. Very useful to know.

**Comment by dyfer** [comment]
How in the world are we getting a wrong peer sample rate reported due to printer/file sharing settings? I think this is something to investigate... maybe a separate issue?

@SynthiaPayne do you have more information in what circumstances such error occurs? Do both machines need to run Windows etc?
